fcrepo-kernel/src/main/java/org/fcrepo/kernel/observer/FedoraEvent.java

```diff
@@ -59,6 +65,7 @@
      */
     public FedoraEvent(final Event e) {
         checkArgument(e != null, "null cannot support a FedoraEvent!");
+        eventID = pidMinter.mintPid();
```

Comment:
Yeah, that's a good point, and I misspoke. What you'd do is have a producer method in the class using `FedoraEvent`. The call to `new` would occur there and the result would be a managed bean.